api/src/geolocate/get.handler.ts

```diff
@@ -0,0 +1,95 @@
+import { APIGatewayProxyEvent, APIGatewayProxyResult } from 'aws-lambda';
+import * as AWS from 'aws-sdk';
+import { RDSDataService } from 'aws-sdk';
+import {
+  ExecuteStatementRequest,
+  FieldList,
+  SqlParametersList,
+} from 'aws-sdk/clients/rdsdataservice';
+
+const SCHEMA = 'public';
+
+/**
+ * Returns all geo identifiers that intersect with a lat, long.
+ * @param rdsService: RDS service to connect to the db.
+ * @param params: with lat and long coordinates sued to query tables.
+ * @param geoid: identifier to retrieve for a table.
+ * @param table: db table to query.
+ */
+async function getGeoDataForLatLong(
+  rdsService: RDSDataService,
+  params: SqlParametersList,
+  geoid: string,
+  table: string,
+): Promise<string | undefined> {
+  const executeParams: ExecuteStatementRequest = {
+    database: process.env.DATABASE_NAME ?? 'postgres',
+    resourceArn: process.env.RESOURCE_ARN ?? '',
+    schema: SCHEMA,
+    secretArn: process.env.CREDENTIALS_SECRET ?? '',
+    sql: `SELECT ${geoid}
+          FROM ${table}
+          WHERE geom && ST_SetSRID(ST_Point(:long, :lat), 4326)
+          LIMIT 1`,
+    parameters: params,
+  };
+  const results = await rdsService.executeStatement(executeParams).promise();
+  const geoids = results.records?.map((record: FieldList) => record[0].stringValue) ?? [];
+  return geoids[0];
+}
+
+export const handler = async (event: APIGatewayProxyEvent): Promise<APIGatewayProxyResult> => {
```

Comment:
The `APIGatewayProxyEvent` is broad and doesn't include our param types. Let's define the request interface here, similar to `GeolocateApiResponse`. E.g.
```
interface GeolocateApiRequest {
  lat: number;
  long: number;
}
```

Reply:
The path params are in the event so I can't change the argument to the handler

Reply:
I didn't realize you could override that way. Updated